packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-transform-job.ts

```diff
@@ -173,6 +181,23 @@ export class SageMakerCreateTransformJob extends sfn.TaskStateBase {
     };
   }
 
+  private renderModelClientOptions(config: ModelClientOptions): { [key: string]: any } {
+    const retries = config.invocationsMaxRetries;
+    if (retries? (retries < 0 || retries > 3): false) {
```

Comment:
these values may be represented as tokens (i.e. if they're passed in as parameters to the stack). we can only validate if they're not tokens, so we would want to add a check that `!Token.isUnResolved(retries)` before proceeding with the check

checkout `Token.isUnresolved` in the repo for a few examples.
Same comment applies to the timeout.